docs/content/developer/tutorials/sponsored-txs.mdx

```diff
@@ -0,0 +1,365 @@
+
+
+
+
+### Second Compononent: Gas Station Server
+
+The Gas Station server is responsible for verifying the user's eligibility for the free trial, building the transaction, and signing it. The server's response will include the transaction and the sponsor's signature.
+We will use [axum](https://docs.rs/axum/latest/axum/) as the web framework for the Gas Station server. However, you can use any web framework you are comfortable with.
+
+To start, create a new rust project called `gas_station_server` by running the following command:
+
+```bash
+cargo new gas_station_server
+```
+You can now open the `gas_station_server` directory with your IDE of choice and add the following dependencies to your `Cargo.toml` file (we will see their usage as we go through the code):
+```
+
+```toml reference
+https://github.com/iota-community/sponsored-transactions-demo/blob/main/backend/Cargo.toml#L7-L23
+```
+
+The Gas Station server will have the following `main` function in the `src/main.rs` file:
+
+:::info
+This implementation of a gas station server has a simple implementation of shared state to keep track of addresses in memory. In a real-world application, you would want to use a database to store this information.
+:::
+
+```rust reference
+https://github.com/iota-community/sponsored-transactions-demo/blob/main/backend/src/main.rs#L124-L150
+```
+
+
+We technically have two routes in the Gas Station server:
+1- The root route `/` will return a welcome message.
+2- The `/sign_and_fund_transaction` route will handle the user's request to sponsor the free trial. This route will call the `sign_and_fund_transaction` function, which we will implement next.
+
+:::danger No data persistence
+In a real-world application, you would want to use a database to store the shared state information. However, for simplicity, we are using a shared state in memory. 
+This means that if the server restarts, the shared state will be lost.
+:::
+
+Next, we will implement the `sign_and_fund_transaction` function, which will be called when the user sends a request to the `/sign_and_fund_transaction` route. This function will verify the user's eligibility, build the transaction, sign it, and return the transaction and the sponsor's signature.
+
+
+```rust reference
+https://github.com/iota-community/sponsored-transactions-demo/blob/main/backend/src/main.rs#L63-L122
+```
+
+The function `sign_and_fund_transaction` will:
+1- Check if the user has already requested funds. If they have, it will return a conflict status.
+2- Add the user's address to the set of addresses that have requested funds.
+3- Build a signed and funded transaction using the `sign_and_fund_transaction` util function, which we will implement next.
+4- Increment the total sponsored fees by the gas price. This is used for tracking by the sponsor.
+5- Respond with the transaction bytes and the sponsor's signature.
```

Comment:
```suggestion
1. Check if the user has already requested funds. If they have, it will return a conflict status.
2. Add the user's address to the set of addresses that have requested funds.
3. Build a signed and funded transaction using the `sign_and_fund_transaction` util function, which we will implement next.
4. Increase the total sponsored fees by the gas price. The sponsor uses this to track expenses.
5. Respond with the transaction bytes and the sponsor's signature.
```